### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/RoslynDataImporter.java [hunk 1]
```diff
@@ -26,17 +26,22 @@
...
 @ScannerSide
 public class RoslynDataImporter {
-  public void importRoslynReport(List<Path> reportPaths, final SensorContext context, Map<String, List<RuleKey>> activeRoslynRulesByPartialRepoKey) {
+  private static final Logger LOG = Loggers.get(RoslynDataImporter.class);
+
+  public void importRoslynReports(List<Path> reportPaths, final SensorContext context, Map<String, List<RuleKey>> activeRoslynRulesByPartialRepoKey) {
     Map<String, String> repositoryKeyByRoslynRuleKey = getRepoKeyByRoslynRuleKey(activeRoslynRulesByPartialRepoKey);
     SarifParserCallback callback = new SarifParserCallbackImpl(context, repositoryKeyByRoslynRuleKey);
 
+    LOG.info("Importing Roslyn {} {}", reportPaths.size(), pluralize("report", reportPaths.size()));
```
Comment (duncanp-sonar) [inline sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/RoslynDataImporter.java]: Message formatting: won't this return e.g. _Importing Roslyn 2 reports_?

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractConfiguration.java [hunk 2]
```diff
@@ -113,24 +117,25 @@ private String getAnalyzerReportDir() {
...
           return false;
         }
 
-        ifNotSilent(silent, () -> LOG.debug("Analyzer working directory contains " + count + " .pb file(s)"));
+        ifNotSilent(silent, () -> LOG.debug("Analyzer working directory '{}' contains {} .pb file(s)", path, count));
         return true;
       }
     } catch (IOException e) {
-      throw new IllegalStateException("Could not check for .pb files in " + analyzerOutputDir.toAbsolutePath(), e);
+      throw new IllegalStateException("Could not check for .pb files in '" + path + "'", e);
```
Comment (duncanp-sonar) [inline sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractConfiguration.java]: Looks like you've changed all of the other messages to not use string concatenation, except this one.

Comment (dbmeneses) [reply sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractConfiguration.java]: good catch

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/RoslynDataImporter.java (hunk 1 above)
Comment (dbmeneses) [reply sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/RoslynDataImporter.java]: good catch

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/RoslynDataImporter.java [hunk 3]
```diff
@@ -55,4 +60,11 @@ public void importRoslynReport(List<Path> reportPaths, final SensorContext conte
     }
     return repositoryKeyByRoslynRuleKey;
   }
+
+  private static String pluralize(String s, int count) {
```
Comment (Evangelink) [inline sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/RoslynDataImporter.java]: Just out of curiosity don't you have any built-in method for this?

Comment (dbmeneses) [reply sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/RoslynDataImporter.java]: No

**Pull request merged**